nsq/async.py

```diff
@@ -468,6 +468,9 @@ def _on_data(self, data, **kwargs):
             self.in_flight += 1
 
             message = protocol.decode_message(data)
+            if self.msg_timeout:
+                message._set_timeout(self)
```

Comment:
Accessing the protected method here isn't ideal. Trying to convey to the user that `_set_timeout` shouldn't be called in their implementations.